### PR TITLE
Test floppy disks actually exist

### LINKED ISSUE #3678: Packer validate doesn't test existence of floppy files (though it does scripts and templates)
When running ```packer validate``` with a template that requires floppy files, it tests the presence of any scripts referenced in the scripts variable, and it also tests the existance of any Vagrant template files, but it does not verify the existence of the floppy files which results in a failing build run even if the template definitely requires them and the validate returned successfully.

- Packer Version = 0.10.1
- Host platform = Windows 10 (PowerShell as cmd interpreter)
- The _simplest example template and scripts_ needed to reproduce the bug.
```git clone https://github.com/boxcutter/ubuntu && mv http/preseed.cfg http/preseed.cfg.bak && packer validate -var-file ubuntu16.04-desktop.json ubuntu.json && packer build -var-file ubuntu16.04-desktop.json ubuntu.json```

### PR DESCRIPTION
**Pull request by ricardclau**
Trying to address bug reported at https://github.com/mitchellh/packer/issues/3678

I just added the check in the vmware builder as a proof of concept, do you want to add a similar snippet everywhere floppies exist? I assume tests would be helpful as well but just wanted to check if I am on the right path first :)

I have of course validated that bad floppy disks were not detected before and they are with this patch. In fact, first commit was failing as the floppies did not exist!

Closes #3678 (when it is finished)

### CODE REVIEW

File: builder/vmware/iso/builder.go [hunk 1]
```diff
@@ -117,6 +117,13 @@ func (b *Builder) Prepare(raws ...interface{}) ([]string, error) {
 		b.config.FloppyFiles = make([]string, 0)
 	}
 
+	for _, path := range b.config.FloppyFiles {
+		if _, err := os.Stat(path); err != nil {
+			errs = packer.MultiErrorAppend(errs,
+				fmt.Errorf("Bad Floppy disk '%s': %s", path, err))
```
Comment (rickard-von-essen) [inline builder/vmware/iso/builder.go]: The working can be slightly improved to something like _Bad floppy disk file:..._.

**Comment by rickard-von-essen** [comment]
At a glance this looks good. I'll like if this is done in all the local builders, virtualbox, parallels, vmware, qemu. This might be a good candidate for a common utility function.

**Comment by ricardclau** [comment]
Oh totally, I just wanted to double check I was on the right path, thanks for the initial review

Will ping you again once all the "floppy aware" builders are done and will export the check to a common utility function

**Comment by ricardclau** [comment]
Ping @rickard-von-essen I think this covers all affected providers and tests now cover them all

Do you guys have any integration environment where you test all the builders with valid templates? I am familiar with virtualbox and vmware but have never used qemu or parallels and it might be worth testing all these in a functional way as well

**Comment by ricardclau** [comment]
oh AppVeyor is failing but not sure if this is related to this PR... any clues? not really familiar with it I am afraid :(

**Comment by rickard-von-essen** [comment]
You can ignore  AppVeyor

**Comment by ricardclau** [comment]
Validated the PR with the vmware-iso provisioner using both Windows with 2 floppies and Ubuntu with no floppies. All behaviours were as expected and floppies paths are validate using `packer validate`

You can check my examples at:

https://github.com/ricardclau/packer-demos/blob/master/templates/windows-server-2012/virtualbox-2012R2-servicebox.json
https://github.com/ricardclau/packer-demos/blob/master/templates/ubuntu-1404/virtualbox-ubuntu.json

**Comment by ricardclau** [comment]
Ping @cbednarski @rickard-von-essen anything else needed here?

**Comment by ricardclau** [comment]
Sorry to be a pain, ping @cbednarski @rickard-von-essen anything else you need to check here?

**Comment by rickard-von-essen** [comment]
Thanks for the PR and for your patient!

**Comment by ricardclau** [comment]
No worries @rickard-von-essen, thanks for your work in Packer and glad this finally got through!

**Pull request merged**